src/main/scala/is/hail/methods/LDMatrix.scala

```diff
@@ -0,0 +1,49 @@
+package is.hail.methods
+
+import is.hail.utils._
+import is.hail.stats.ToNormalizedIndexedRowMatrix
+import is.hail.variant.{Variant, VariantDataset}
+import is.hail.utils.richUtils.RichIndexedRowMatrix._
+import org.apache.spark.mllib.linalg.distributed.{IndexedRow, IndexedRowMatrix}
+
+
+object LDMatrix {
+
+  /**
+    * Computes the LD matrix for the given VDS.
+    * @param vds The VDS to compute LD matrix of.
+    * @return An LDMatrix.
+    */
+  def apply(vds : VariantDataset): LDMatrix = {
+    val nSamples = vds.nSamples
+    val originalVariants = vds.variants.collect().sorted
+
+    val normalizedIRM = ToNormalizedIndexedRowMatrix(vds)
+    normalizedIRM.rows.persist()
+    val variantIndices = normalizedIRM.rows.map{case IndexedRow(idx, _) => idx.toInt}.collect().sorted
+    val variantsUsed = variantIndices.map(idx => originalVariants(idx))
+
+    info(s"Computing LDMatrix with ${variantsUsed.length} variants and $nSamples samples")
```

Comment:
(perhaps you should also note the number that were removed due to having no variance...people may be confused to see number of variants less than vds.countVariants())